ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/JaasAuthProvider.java

```diff
@@ -128,7 +130,16 @@ private void getUser(
     // We do the actual authorization here not in the User class
     final boolean authorized = validateRoles(lc, allowedRoles);
 
-    promise.complete(new JaasUser(username, password, authorized));
+    // if the subject from the login context is already a KsqlPrincipal, use the subject
+    // directly rather than creating a new one
+    final Optional<KsqlPrincipal> ksqlPrincipal = lc.getSubject().getPrincipals().stream()
```

Comment:
Do we ever expect more than 1 principal here?

Reply:
The ksql codebase is only really set up for a single principal based on the interfaces used throughout the engine, so I think it's fine to only expect one.